mani_skill/utils/scene_builder/scene_builder.py

```diff
@@ -1,60 +1,91 @@
 """NOTE: This scene builder class API is a WIP still. Temporarily used for managing a few pre-built scenes"""
+
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Dict, List
+from typing import TYPE_CHECKING, Any, Dict, List, Union, Optional
+from functools import cached_property
 
-import sapien
 import torch
+from gymnasium import spaces
 
 if TYPE_CHECKING:
     from mani_skill.envs.sapien_env import BaseEnv
-
-from mani_skill.utils.structs.actor import Actor
+from mani_skill.utils.structs import Actor, Articulation
+from mani_skill.utils.structs.types import Array
 
 
 class SceneBuilder:
     """Base class for defining scene builders that can be reused across tasks"""
 
+    """Env which scenebuilder will build in."""
     env: BaseEnv
-    _scene_objects: List[Actor] = []
-    _movable_objects: List[Actor] = []
+
+    """Robot init qpos noise"""
+    robot_init_qpos_noise: float = 0.02
+
+    """Whether this scene builder will add its own lighting when build is called. If False, ManiSkill will add some default lighting"""
     builds_lighting: bool = False
-    """Whether this scene builder will add it's own lighting when build is called. If False, ManiSkill will add some default lighting"""
-    scene_configs: List[Any] = None
-    """List of scene configuration information that can be used to construct scenes. Can be simply a path to a json file or a dictionary"""
+
+    """
+    **Optional** list of scene configuration information that can be used to build/init scenes. Can be a dictionary, a path to a json file, or some other data.
+    Some scenes will need to load config data, while others might not.
+    """
+    build_configs: Optional[List[Any]] = None
+    init_configs: Optional[List[Any]] = None
+
+    """
+    Dictionaries mapping names to scene objects, movable objects, and articulations for easy reference.
+    """
+    scene_objects: Optional[Dict[str, Actor]] = None
+    movable_objects: Optional[Dict[str, Actor]] = None
+    articulations: Optional[Dict[str, Articulation]] = None
```

Comment:
can you document the scene objects vs movable objects difference?

moreover comments should be below the variable otherwise it's not shown correctly in editors/docs

Reply:
added documentation and also moved comments to under the variable